File: stubs/csharp/src/BeastMode.RmDdd/Core/ModuleHealth.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace BeastMode.RmDdd.Core
{
    /// <summary>
    /// Comprehensive module health information.
    /// </summary>
    public class ModuleHealth
    {
        public ModuleStatus Status { get; }
        public string Message { get; }
        public IReadOnlyList<ModuleCapability> Capabilities { get; }
        public DomainHealth? DomainHealth { get; }
        public IReadOnlyDictionary<string, object> HealthIndicators { get; }
        public DateTimeOffset Timestamp { get; }
        
        public ModuleHealth(
            ModuleStatus status,
            string message,
            IEnumerable<ModuleCapability> capabilities,
            DomainHealth? domainHealth = null,
            IDictionary<string, object>? healthIndicators = null)
        {
            Status = status;
            Message = message ?? throw new ArgumentNullException(nameof(message));
            Capabilities = capabilities?.ToList().AsReadOnly() ?? throw new ArgumentNullException(nameof(capabilities));
            DomainHealth = domainHealth;
            HealthIndicators = (healthIndicators ?? new Dictionary<string, object>()).ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
            Timestamp = DateTimeOffset.UtcNow;
        }
        
        /// <summary>
        /// Gets whether the module is healthy
        /// </summary>
        public bool IsHealthy => Status == ModuleStatus.Available;
        
        public override bool Equals(object? obj)
        {
            if (obj is not ModuleHealth other) return false;
            
            return Status == other.Status &&
                   Message == other.Message &&
                   Capabilities.SequenceEqual(other.Capabilities) &&
                   Equals(DomainHealth, other.DomainHealth) &&
                   HealthIndicators.SequenceEqual(other.HealthIndicators);
        }
        
        public override int GetHashCode()
        {
            return HashCode.Combine(Status, Message, Capabilities, DomainHealth, HealthIndicators);
        }
        
        public override string ToString()
        {
            return $"ModuleHealth {{ Status = {Status}, Message = '{Message}', Capabilities = {Capabilities.Count}, Timestamp = {Timestamp} }}";
        }
    }
}

File: stubs/java/src/main/java/com/beastmode/rmddd/core/ModuleHealth.java
package com.beastmode.rmddd.core;

import java.time.Instant;
import java.util.List;
import java.util.Map;
import java.util.Objects;

/**
 * Comprehensive module health information.
 */
public class ModuleHealth {
    
    private final ModuleStatus status;
    private final String message;
    private final List<ModuleCapability> capabilities;
    private final DomainHealth domainHealth;
    private final Map<String, Object> healthIndicators;
    private final Instant timestamp;
    
    public ModuleHealth(ModuleStatus status, 
                       String message, 
                       List<ModuleCapability> capabilities,
                       DomainHealth domainHealth,
                       Map<String, Object> healthIndicators) {
        this.status = Objects.requireNonNull(status, "Status cannot be null");
        this.message = Objects.requireNonNull(message, "Message cannot be null");
        this.capabilities = Objects.requireNonNull(capabilities, "Capabilities cannot be null");
        this.domainHealth = domainHealth;
        this.healthIndicators = Objects.requireNonNull(healthIndicators, "Health indicators cannot be null");
        this.timestamp = Instant.now();
    }
    
    public ModuleStatus getStatus() {
        return status;
    }
    
    public String getMessage() {
        return message;
    }
    
    public List<ModuleCapability> getCapabilities() {
        return capabilities;
    }
    
    public DomainHealth getDomainHealth() {
        return domainHealth;
    }
    
    public Map<String, Object> getHealthIndicators() {
        return healthIndicators;
    }
    
    public Instant getTimestamp() {
        return timestamp;
    }
    
    public boolean isHealthy() {
        return status == ModuleStatus.AVAILABLE;
    }
    
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ModuleHealth that = (ModuleHealth) o;
        return status == that.status &&
               Objects.equals(message, that.message) &&
               Objects.equals(capabilities, that.capabilities) &&
               Objects.equals(domainHealth, that.domainHealth) &&
               Objects.equals(healthIndicators, that.healthIndicators);
    }
    
    @Override
    public int hashCode() {
        return Objects.hash(status, message, capabilities, domainHealth, healthIndicators);
    }
    
    @Override
    public String toString() {
        return "ModuleHealth{" +
               "status=" + status +
               ", message='" + message + '\'' +
               ", capabilities=" + capabilities +
               ", domainHealth=" + domainHealth +
               ", healthIndicators=" + healthIndicators +
               ", timestamp=" + timestamp +
               '}';
    }
}